Description: Remove det() function

 Unfortunately the file determinant.hpp did not have adequate
 copyright information, however it is not used by Siconos.
 .
 siconos (4.2.0+git20181026.0ee5349+dfsg-1) unstable; urgency=medium
 .
Author: Stephen Sinclair <radarsat1@gmail.com>

diff --git a/kernel/src/utils/SiconosAlgebra/SimpleMatrix.hpp b/kernel/src/utils/SiconosAlgebra/SimpleMatrix.hpp
index e0e1ebf..037e90b 100644
--- a/kernel/src/utils/SiconosAlgebra/SimpleMatrix.hpp
+++ b/kernel/src/utils/SiconosAlgebra/SimpleMatrix.hpp
@@ -427,11 +427,6 @@ public:
    */
   void normInfByColumn(SP::SiconosVector vIn) const;
 
-  /** compute the determinant of the matrix (use LU factorization)
-   *  \return a double
-   */
-  double det() const;
-
   /** display data on standard output
    */
   void display() const;
diff --git a/kernel/src/utils/SiconosAlgebra/SimpleMatrixMisc.cpp b/kernel/src/utils/SiconosAlgebra/SimpleMatrixMisc.cpp
index 9bf88b7..1bd68d6 100644
--- a/kernel/src/utils/SiconosAlgebra/SimpleMatrixMisc.cpp
+++ b/kernel/src/utils/SiconosAlgebra/SimpleMatrixMisc.cpp
@@ -20,7 +20,6 @@
 #include <boost/numeric/ublas/matrix_proxy.hpp>
 
 
-#include "determinant.hpp"
 #include "SiconosVector.hpp"
 #include "SimpleMatrix.hpp"
 #include "BlockMatrixIterators.hpp"
@@ -68,27 +67,6 @@ void SimpleMatrix::normInfByColumn(SP::SiconosVector vIn) const
   else
     RuntimeException::selfThrow("SimpleMatrix::normInfByColumn: not implemented for data other than DenseMat");
 }
-//=======================
-//       determinant
-//=======================
-
-double SimpleMatrix::det() const
-{
-  if (_num == 1)
-    return determinant(*mat.Dense);
-  else if (_num == 2)
-    return determinant(*mat.Triang);
-  else if (_num == 3)
-    return determinant(*mat.Sym);
-  else if (_num == 4)
-    return determinant(*mat.Sparse);
-  else if (_num == 5)
-    return determinant(*mat.Banded);
-  else if (_num == 6)
-    return 0;
-  else // if(_num==7)
-    return 1;
-}
 
 
 void SimpleMatrix::trans()
